java/mvn/cloudbuild.yaml

```diff
@@ -2,8 +2,45 @@
 # $ gcloud container builds submit . --config=cloudbuild.yaml
 
 steps:
+# Build the java / maven matrix.
 - name: 'gcr.io/cloud-builders/docker'
-  args: ['build', '--file=Dockerfile.8-jdk-alpine', '--build-arg=MAVEN_VERSION=3.5.0', '--tag=gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-8', '.']
+  args:
+  - 'build'
+  - '--build-arg=BASE_IMAGE=openjdk:8-jdk-alpine'
+  - '--build-arg=MAVEN_VERSION=3.3.9'
+  - '--build-arg=SHA=6e3e9c949ab4695a204f74038717aa7b2689b1be94875899ac1b3fe42800ff82'
+  - '--tag=gcr.io/$PROJECT_ID/java/mvn:3.3.9-jdk-8'
+  - '.'
+  wait_for: ['-']
+- name: 'gcr.io/cloud-builders/docker'
+  args:
+  - 'build'
+  - '--build-arg=BASE_IMAGE=openjdk:7-jdk-alpine'
+  - '--build-arg=MAVEN_VERSION=3.3.9'
+  - '--build-arg=SHA=6e3e9c949ab4695a204f74038717aa7b2689b1be94875899ac1b3fe42800ff82'
+  - '--tag=gcr.io/$PROJECT_ID/java/mvn:3.3.9-jdk-7'
+  - '.'
+  wait_for: ['-']
+- name: 'gcr.io/cloud-builders/docker'
+  args:
+  - 'build'
+  - '--build-arg=BASE_IMAGE=openjdk:8-jdk-alpine'
+  - '--build-arg=MAVEN_VERSION=3.5.0'
+  - '--build-arg=SHA=beb91419245395bd69a4a6edad5ca3ec1a8b64e41457672dc687c173a495f034'
+  - '--tag=gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-8'
+  - '.'
+  wait_for: ['-']
+- name: 'gcr.io/cloud-builders/docker'
+  args:
+  - 'build'
+  - '--build-arg=BASE_IMAGE=openjdk:7-jdk-alpine'
+  - '--build-arg=MAVEN_VERSION=3.5.0'
+  - '--build-arg=SHA=beb91419245395bd69a4a6edad5ca3ec1a8b64e41457672dc687c173a495f034'
+  - '--tag=gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-7'
+  - '.'
+  wait_for: ['-']
+
+# Tag a default builder version.
 - name: 'gcr.io/cloud-builders/docker'
   args: ['tag', 'gcr.io/$PROJECT_ID/java/mvn:3.5.0-jdk-8', 'gcr.io/$PROJECT_ID/java/mvn']
```

Comment:
This step only needs to `waitFor` the step that builds the `3.5.0-jdk-8` image, right?